juspay/hyperswitch
# feat(core): add profile  level config to toggle extended card bin

## Type of Change



## Description
Add profile level config to toggle the display for the extended card bin in Payment Response .

### Additional Changes





## Motivation and Context



## How did you test it?
 

- Create a MA and an MCA
- Make a payment with the toggle for that profile_id as false, as it is default
- In the response the card_bin_extended field will be null
<img width="1049" alt="Screenshot 2024-04-24 at 4 13 09 AM" src="https://github.com/juspay/hyperswitch/assets/55580080/d1dd3166-de79-49fd-8c03-f115125db0a4">

- Now set the toggle for that profile_id as true by hitting the following curl:
```
curl --location 'http://localhost:8080/configs/{profile_id}_enable_extended_card_bin' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: {{admin_api_key}}' \
--data '{
    "key":"{profile_id}_enable_extended_card_bin",
    "value":"true"
}'
```
> <img width="1210" alt="Screenshot 2024-05-02 at 1 06 08 PM" src="https://github.com/juspay/hyperswitch/assets/55580080/89341370-de38-4280-a527-72ca8dc13549">


> > The config will also be stored in the database, to check you can run the following query
> ```
>  SELECT * FROM configs WHERE key = '{profile_id}_enable_extended_card_bin';
>  ```

> <img width="1210" alt="Screenshot 2024-05-02 at 1 07 48 PM" src="https://github.com/juspay/hyperswitch/assets/55580080/c020a3cf-3154-45e6-a079-04a7490a8930">


- Create another payment , now in the Response card_bin_extended field won't be null, as the extended card bin config has been set to true
<img width="1049" alt="Screenshot 2024-04-24 at 4 14 50 AM" src="https://github.com/juspay/hyperswitch/assets/55580080/aab6d569-bfa0-4a3e-bd3b-bebbb3525fbc">





## Checklist
